catalog/from-k8s/resource.go

```diff
@@ -277,52 +277,58 @@ func (t *ServiceResource) generateRegistrations(key string) {
 	}
 
 	// Determine the default port and set port annotations
+	var endpointPortName string
+	var overridePortNumber int
 	if len(svc.Spec.Ports) > 0 {
-		// Create port variable, defaults to 0
 		var port int
-
-		// Flag identifying whether the service is of NodePort type
 		isNodePort := svc.Spec.Type == apiv1.ServiceTypeNodePort
 
 		// If a specific port is specified, then use that port value
-		target, ok := svc.Annotations[annotationServicePort]
+		portAnnotation, ok := svc.Annotations[annotationServicePort]
 		if ok {
-			if v, err := strconv.ParseInt(target, 0, 0); err == nil {
+			if v, err := strconv.ParseInt(portAnnotation, 0, 0); err == nil {
 				port = int(v)
```

Comment:
This is the datatype on the consul api struct so we can't change it: https://github.com/hashicorp/consul/blob/master/api/agent.go#L71

Sometimes ports in consul get set to `-1` to state that they're disable so that's probably why they use `int`. In the case of a service this wouldn't make any sense though.

Reply:
I see, thanks for clarifying.